internal/ingress/controller/controller.go

```diff
@@ -1102,6 +1120,57 @@ func (n *NGINXController) createServers(data []*extensions.Ingress,
 	return servers
 }
 
+// match and merge virtual backends into their respective real backends
+func mergeVirtualBackends(ing *extensions.Ingress, anns *annotations.Ingress, upstreams map[string]*ingress.Backend,
+	servers map[string]*ingress.Server) {
+
+	for _, rule := range ing.Spec.Rules {
+		for _, path := range rule.HTTP.Paths {
+			upsName := fmt.Sprintf("%v-%v-%v",
+				ing.Namespace,
+				path.Backend.ServiceName,
+				path.Backend.ServicePort.String())
```

Comment:
could you create a utility function for this and DRY it everywhere else in this package as well?